examples/vector_databases/redis/redis-hybrid-query-examples.ipynb

```diff
@@ -0,0 +1,862 @@
+    "3. Create the index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "f894b911",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Constants\n",
+    "INDEX_NAME = \"product_embeddings\"           # name of the search index\n",
+    "PREFIX = \"doc\"                            # prefix for the document keys\n",
+    "DISTANCE_METRIC = \"L2\"                # distance metric for the vectors (ex. COSINE, IP, L2)\n",
+    "NUMBER_OF_VECTORS = len(df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "15db8380",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define RediSearch fields for each of the columns in the dataset\n",
+    "name = TextField(name=\"productDisplayName\")\n",
+    "category = TagField(name=\"masterCategory\")\n",
+    "articleType = TagField(name=\"articleType\")\n",
+    "gender = TagField(name=\"gender\")\n",
+    "season = TagField(name=\"season\")\n",
+    "year = NumericField(name=\"year\")\n",
+    "text_embedding = VectorField(\"product_vector\",\n",
+    "    \"FLAT\", {\n",
+    "        \"TYPE\": \"FLOAT32\",\n",
+    "        \"DIM\": 1536,\n",
+    "        \"DISTANCE_METRIC\": DISTANCE_METRIC,\n",
+    "        \"INITIAL_CAP\": NUMBER_OF_VECTORS,\n",
+    "    }\n",
+    ")\n",
+    "fields = [name, category, articleType, gender, season, year, text_embedding]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "3658693c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Index already exists\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Check if index exists\n",
+    "try:\n",
+    "    redis_client.ft(INDEX_NAME).info()\n",
+    "    print(\"Index already exists\")\n",
+    "except:\n",
+    "    # Create RediSearch Index\n",
+    "    redis_client.ft(INDEX_NAME).create_index(\n",
+    "        fields = fields,\n",
+    "        definition = IndexDefinition(prefix=[PREFIX], index_type=IndexType.HASH)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "775c15b4",
+   "metadata": {},
+   "source": [
+    "## Generate OpenAI Embeddings and Load Documents into the Index\n",
+    "\n",
+    "Now that we have a search index, we can load documents into it. We will use the dataframe containing the styles dataset loaded previously. In Redis, either the HASH or JSON (if using RedisJSON in addition to RediSearch) data types can be used to store documents. We will use the HASH data type in this example. The cells bellow will show how to get OpenAI embeddings for the different products and load documents into the index."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "0d791186",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def index_documents(client: redis.Redis, prefix: str, documents: pd.DataFrame):\n",
```

Comment:
best to use a redis pipeline for this (in batches of ~500) to demonstrate best practice for writing large quantities of data